File: Remove Duplicates from Sorted Array II.java
public class Solution {
    public int removeDuplicates(int[] A) {
        int w = 0;
        Map<Integer, Integer> count = new HashMap<Integer, Integer>();
        for (int r = 0; r < A.length; r++) {
            if (!count.containsKey(A[r]))
                count.put(A[r], 0);
            if (count.get(A[r]) == 2)
                continue;
            count.put(A[r], count.get(A[r])+1);
            A[w++] = A[r];
        }
        return w;
    }
}

File: Remove Duplicates from Sorted Array II.py
class Solution:
	# @param A a list of integers
	# @return an integer
	def removeDuplicates(self, A):
		w = 0
		count = {}
		for r in xrange(len(A)):
			if (not count.has_key(A[r])):
				count[A[r]] = 0
			if (count[A[r]] == 2):
				continue
			count[A[r]] += 1
			A[w] = A[r]
			w += 1
		return w